curp/src/server/raw_curp/log.rs

```diff
@@ -22,17 +24,34 @@ pub(super) struct Log<C: Command> {
     pub(super) commit_index: usize,
     /// Index of highest log entry applied to state machine
     pub(super) last_applied: usize,
+    /// Tx to send log entries to persist task
+    log_tx: mpsc::UnboundedSender<LogEntry<C>>,
+}
+
+impl<C: Command> Debug for Log<C> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Log")
+            .field("entries", &self.entries)
+            .field("base_index", &self.base_index)
+            .field("commit_index", &self.commit_index)
+            .field("last_applied", &self.last_applied)
+            .finish()
+    }
 }
 
-impl<C: Command> Log<C> {
+impl<C: 'static + Command> Log<C> {
     /// Create a new log
-    pub(super) fn new() -> Self {
+    pub(super) fn new(
+        log_tx: mpsc::UnboundedSender<LogEntry<C>>,
+        entries: Vec<LogEntry<C>>,
+    ) -> Self {
         Self {
-            entries: vec![], // a fake log[0]
+            entries, // a fake log[0]
             commit_index: 0,
```

Comment:
why we didn't recover those indexes?

Reply:
Last applied should be maintained by command executor while commit index can be fetched from the leader as soon as it recovers